### ISSUE TITLE
Confusing Scaleway parameter names at odds with Scaleway's use

### ISSUE THREAD

**Issue by OJFord** [hashicorp/terraform#10815]
### Terraform Version
`0.8.1`

### Affected Resource(s)
- (provider) `scaleway`

The required parameters `organization` and `access_key` are *very* confusing:
- TF 'organization' is SW 'access_key'
- TF 'access_key' is SW 'token'

Is there a reason Terraform does not just use 'access_key' and 'token'? (This would also benefit from avoiding localisation issues - Commonwealth users are bound to mess it up as I did, which results in a confusing prompt to enter information you thought you already had!)

**Comment by nicolai86** [comment]
Hey @OJFord 

thanks for reaching out! I agree that this is inconsistent.

To some degree the inconsistency was inherited from Scaleway itself. E.g. the Scaleway UI calls `organization` access key, and the Scaleway CLI calls it organization again. This is also reflected in the `~/.scwrc` file which the official scaleway CLI generates:

```
{
  "organization": "7bffff27-4ddd-0000-0000-1aaaaaaa5889",
  "token": "36dfff9f-0000-4ddd-0000-a57007aaaa30",
  "version": "v1.11.1"
}
```

Since we rely on the official Scaleway SDK I think it would be best to mirror their names, in which case we should modify the provider to allow `SCALEWAY_TOKEN` to be used instead of `SCALEWAY_ACCESS_KEY`, and also introduce a `token` parameter as a replacement for `access_key`.

It's a tiny change, assuming we leave the current configuration in place for backwards compatibility.
I would love a PR for this which adjusts the naming :)

**Comment by OJFord** [comment]
That sounds good, it at least removes the contradiction.

Docs could presumably then mention both organisation and access key aliases to head off further confusion?

Happy to look into producing a PR tomorrow if it ought to be as straightforward as I'd naïvely imagine - I've no familiarity with the codebase at all.

**Comment by nicolai86** [comment]
Generally speaking you need to modify two files:

- the Scaleway provider definition (`builtin/providers/scaleway/provider.go`)
- the Scaleway provider documentation (`website/source/docs/providers/scaleway/index.html.markdown`)

if you need any assistance for your PR let me know.

**Comment by stack72** [comment]
Closed via #10874
